common/client-core/src/client/topology_control/nym_api_provider.rs

```diff
@@ -135,31 +136,31 @@ impl NymApiTopologyProvider {
         } else {
             // if we're not using extended topology, we're only getting active set mixnodes and gateways
 
-            let mixnodes_fut = self
-                .validator_client
-                .get_all_basic_active_mixing_assigned_nodes_with_metadata();
-
-            // TODO: we really should be getting ACTIVE gateways only
-            let gateways_fut = self
-                .validator_client
-                .get_all_basic_entry_assigned_nodes_with_metadata();
+            // Get ALL bonded nodes and filter for both mixnodes and gateways
+            // This ensures consistent metadata between both calls
+            let all_nodes_fut = self.validator_client.get_all_basic_nodes_with_metadata();
 
-            let (rewarded_set, mixnodes_res, gateways_res) =
-                futures::try_join!(rewarded_set_fut, mixnodes_fut, gateways_fut)
-                    .inspect_err(|err| {
-                        error!("failed to get network nodes: {err}");
-                    })
-                    .ok()?;
+            let (rewarded_set, all_nodes_res) = futures::try_join!(rewarded_set_fut, all_nodes_fut)
+                .inspect_err(|err| {
+                    error!("failed to get network nodes: {err}");
+                })
+                .ok()?;
 
-            let metadata = mixnodes_res.metadata;
-            let mixnodes = mixnodes_res.nodes;
+            let metadata = all_nodes_res.metadata;
+            let all_nodes = all_nodes_res.nodes;
 
-            if !gateways_res.metadata.consistency_check(&metadata) {
-                warn!("inconsistent nodes metadata between mixnodes and gateways calls! {metadata:?} and {:?}", gateways_res.metadata);
-                return None;
-            }
+            // Filter for active mixing nodes (role=Mixnode)
+            let mixnodes: Vec<_> = all_nodes
+                .iter()
+                .filter(|n| n.supported_roles.mixnode && matches!(n.role, NodeRole::Mixnode { .. }))
+                .cloned()
+                .collect();
 
-            let gateways = gateways_res.nodes;
+            // Filter for entry-capable nodes (supported_roles.entry == true)
+            let gateways: Vec<_> = all_nodes
```

Comment:
you're essentially recreating what nym-api is doing for you when you call `get_all_basic_entry_assigned_nodes_with_metadata` (it already filters via node capabilities)